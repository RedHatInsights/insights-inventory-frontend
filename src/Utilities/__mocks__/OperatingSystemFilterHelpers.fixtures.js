export const availableVersions = [
  {
    groupLabel: 'RHEL 6',
    label: 'RHEL 6.9',
    osName: 'RHEL',
    value: '6.9',
  },
  {
    groupLabel: 'RHEL 6',
    label: 'RHEL 6.10',
    osName: 'RHEL',
    value: '6.10',
  },
  {
    groupLabel: 'RHEL 7',
    label: 'RHEL 7.0',
    osName: 'RHEL',
    value: '7.0',
  },
  {
    groupLabel: 'RHEL 8',
    label: 'RHEL 8.5',
    osName: 'RHEL',
    value: '8.5',
  },
  {
    groupLabel: 'RHEL 8',
    label: 'RHEL 8.10',
    osName: 'RHEL',
    value: '8.10',
  },
  {
    groupLabel: 'CentOS Linux 7',
    label: 'CentOS Linux 7.9',
    osName: 'CentOS Linux',
    value: '7.9',
  },
  {
    groupLabel: 'CentOS Linux 8',
    label: 'CentOS Linux 8.6',
    osName: 'CentOS Linux',
    value: '8.6',
  },
];

export const testValue1 = {
  'RHEL 7': {
    'RHEL 7': true,
    '7.0': true,
    7.1: true,
    7.2: true,
  },
  'RHEL 8': {
    'RHEL 8': true,
    8.5: true,
    '8.10': true,
  },
  'RHEL 9': {
    'RHEL 9': true,
    9.1: true,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': true,
    7.6: true,
    7.9: true,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': true,
    8.3: true,
    8.9: true,
  },
};

/*eslint-disable prettier/prettier*/
export const testValue2 = {
  'RHEL 6': {
    'RHEL 6': false,
    '6.9': true,
    '6.10': false,
  },
  'RHEL 7': {
    'RHEL 7': true,
    '7.0': true,
  },
  'RHEL 8': {
    'RHEL 8': true,
    '8.5': true,
    '8.10': true,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': true,
    '7.9': true,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': false,
    '8.6': false,
  },
};

export const testValue3 = {
  'RHEL 6': {
    'RHEL 6': false,
    '6.9': false,
    '6.10': false,
  },
  'RHEL 7': {
    'RHEL 7': false,
    '7.0': false,
  },
  'RHEL 8': {
    'RHEL 8': false,
    '8.5': false,
    '8.10': false,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': false,
    '7.9': false,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': false,
    '8.6': false,
  },
};

export const testValue4 = {
  'RHEL 6': {
    'RHEL 6': false,
    '6.9': false,
    '6.10': false,
  },
  'RHEL 7': {
    'RHEL 7': false,
    '7.0': false,
  },
  'RHEL 8': {
    'RHEL 8': true,
    '8.5': true,
    '8.10': true,
  },
  'CentOS Linux 7': {
    'CentOS Linux 7': false,
    '7.9': false,
  },
  'CentOS Linux 8': {
    'CentOS Linux 8': false,
    '8.6': false,
  },
};
/*eslint-enable prettier/prettier*/

export const groupSelection1 = [
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.0' },
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.1' },
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.2' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.5' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.10' },
  { osName: 'RHEL', osGroup: 'RHEL 9', value: '9.1' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 7', value: '7.6' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 7', value: '7.9' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 8', value: '8.3' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 8', value: '8.9' },
];

export const groupSelection2 = [
  { osName: 'RHEL', osGroup: 'RHEL 6', value: '6.9' },
  { osName: 'RHEL', osGroup: 'RHEL 7', value: '7.0' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.5' },
  { osName: 'RHEL', osGroup: 'RHEL 8', value: '8.10' },
  { osName: 'CentOS Linux', osGroup: 'CentOS Linux 7', value: '7.9' },
];
