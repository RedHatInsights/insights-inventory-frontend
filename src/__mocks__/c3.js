// eslint-disable-next-line no-undef
module.exports = () => 'c3';

