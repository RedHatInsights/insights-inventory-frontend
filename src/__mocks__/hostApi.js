import { hosts, tags } from '../api/api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios, { onNoMatch: 'throwException' });
export const mockTags = new MockAdapter(tags.axios, { onNoMatch: 'throwException' });
