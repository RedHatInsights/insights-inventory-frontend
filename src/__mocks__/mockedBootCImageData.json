[
    {
        "image": "India Pale Ale",
        "hashCommitCount": 2,
        "systemCount": 3,
        "hashes": [
            {
                "image_digest": "Hazy",
                "hashSystemCount": 2
            },
            {
                "image_digest": "West Coast",
                "hashSystemCount": 1
            }
        ]
    },
    {
        "image": "Belgian",
        "hashCommitCount": 1,
        "systemCount": 1,
        "hashes": [
            {
                "image_digest": "Dubbel",
                "hashSystemCount": 1
            }
        ]
    },
    {
        "image": "Stout",
        "hashCommitCount": 1,
        "systemCount": 1,
        "hashes": [
            {
                "image_digest": "Bourbon Barrell Aged",
                "hashSystemCount": 1
            }
        ]
    }
]