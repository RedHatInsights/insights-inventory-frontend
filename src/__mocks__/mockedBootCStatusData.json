{
    "total": 5,
    "count": 5,
    "page": 1,
    "per_page": 50,
    "results": [
        {
            "id": "6de30d78-f052-41d4-b123-125017443a98",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "7.4",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "India Pale Ale",
                        "image_digest": "Hazy"
                    }
                }
            }
        },
        {
            "id": "08a06e04-afbb-4e3c-b923-91aa1c17acc1",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.4",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "India Pale Ale",
                        "image_digest": "Hazy"
                    }
                }
            }
        },
        {
            "id": "4ef68d04-79fa-42ea-a0fe-b40ee257ff0e",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.5",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "India Pale Ale",
                        "image_digest": "West Coast"
                    }
                }
            }
        },
        {
            "id": "b6197999-06f8-4301-a936-651832aa8b31",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.5",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "Belgian",
                        "image_digest": "Dubbel"
                    }
                }
            }
        },
        {
            "id": "8499b198-5352-4eda-8650-ca09e002fdc1",
            "system_profile": {
                "arch": "x86_64",
                "cpu_flags": [ "fpu" ],
                "yum_repos": [
                    {
                        "name": "Extra Packages for Enterprise Linux 7 - Testing - $basearch",
                        "enabled": false,
                        "gpgcheck": true
                    }
                ],
                "os_release": "8.2",
                "bios_vendor": "SeaBIOS",
                "bios_version": "1.10.2-2.fc27",
                "kernel_modules": [ "veth" ],
                "last_boot_time": "2018-03-14T17:06:10",
                "number_of_cpus": 1,
                "cores_per_socket": 1,
                "enabled_services": [ "auditd" ],
                "bios_release_date": "04\/01\/2014",
                "number_of_sockets": 1,
                "os_kernel_version": "3.10.0",
                "running_processes": [ "qmgr" ],
                "satellite_managed": false,
                "installed_packages": [ "0:GeoIP-1.5.0-11.el7" ],
                "installed_services": [ "arp-ethers" ],
                "network_interfaces": [
                    {
                        "mtu": 65536,
                        "name": "lo",
                        "type": "loopback",
                        "state": "UNKNOWN",
                        "mac_address": "00:00:00:00:00:00",
                        "ipv4_addresses": [
                            "127.0.0.1"
                        ],
                        "ipv6_addresses": [
                            "::1"
                        ]
                    }
                ],
                "infrastructure_type": "virtual",
                "system_memory_bytes": 512073728,
                "infrastructure_vendor": "qemu",
                "system_purpose": {
                    "usage": "Production"
                },
                "bootc_status": {
                    "booted": {
                        "image": "Stout",
                        "image_digest": "Bourbon Barrell Aged"
                    }
                }
            }
        }
    ]
}
