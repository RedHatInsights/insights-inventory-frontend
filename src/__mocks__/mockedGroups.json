{
  "total": 1,
  "count": 1,
  "page": 1,
  "per_page": 50,
  "results": [
    {
      "name": "aute",
      "id": "FC7CDc2A-EbCd-36F5-f2cc-72f7f2fB960F",
      "host_ids": [
        "2Ab5EF47-a25f-eC8D-459D-e48Ca051b1F2",
        "EdE9F657-B7FD-E56c-2E66-6a93ee8aabcC"
      ],
      "created_at": "2009-01-31T23:00:00.0Z",
      "updated_at": "1981-07-06T22:00:00.0Z"
    }
  ]
}
