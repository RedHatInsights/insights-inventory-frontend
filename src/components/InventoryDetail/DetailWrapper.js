import React, { useEffect } from 'react';
import {
    Drawer,
    DrawerPanelContent,
    DrawerContent,
    DrawerContentBody,
    DrawerPanelBody,
    DrawerActions,
    DrawerHead,
    DrawerCloseButton,
    Stack,
    StackItem
} from '@patternfly/react-core';
import PropTypes from 'prop-types';
import { useSelector, useDispatch, useStore } from 'react-redux';
import { toggleDrawer, loadEntity } from '../../store/actions';
import { BasicInfo, SystemIssues } from '../InventoryDetailDrawer';
import FactsInfo from './FactsInfo';

const DetailWrapper = ({ children, hideInvLink, showTags, Wrapper, className, hasAccess, appName, inventoryId, ...props }) => {
    const dispatch = useDispatch();
    const store = useStore();
    const entity = useSelector(({ entityDetails }) => entityDetails?.entity);
    const isExpanded = useSelector(({ entityDetails }) => entityDetails?.isToggleOpened);
    const loaded = useSelector(({ entityDetails }) => entityDetails?.loaded);

    useEffect(() => {
        if (inventoryId && !entity || !(entity?.id === inventoryId)) {
            dispatch(loadEntity(inventoryId, { hasItems: true }, { showTags }));
        }
    }, [entity, inventoryId]);

    return <Drawer
        className={`ins-c-inventory__drawer ${className || ''}`}
        isExpanded={isExpanded}
        onExpand={() => dispatch(toggleDrawer(true))}
        {...props}
    >
        <DrawerContent
            panelContent={
                <DrawerPanelContent>
                    <DrawerHead>
                        <BasicInfo hideInvLink={ hideInvLink } showTags={ showTags } />
                        <DrawerActions>
                            <DrawerCloseButton onClick={() =>  dispatch(toggleDrawer(false))} />
                        </DrawerActions>
                    </DrawerHead>
                    <DrawerPanelBody>
                        <Stack className="ins-c-inventory__drawer--content">
                            <StackItem>
                                <SystemIssues isOpened={isExpanded} />
                            </StackItem>
                            <StackItem isFilled className="ins-c-inventory__drawer--facts">
                                <FactsInfo entity={entity} loaded={loaded} />
                                {
                                    isExpanded &&
                                    loaded &&
                                    Wrapper &&
                                    <Wrapper
                                        store={store}
                                        appName={appName}
                                    />
                                }
                            </StackItem>
                        </Stack>
                    </DrawerPanelBody>
                </DrawerPanelContent>
            }
        >
            <DrawerContentBody>
                {children}
            </DrawerContentBody>
        </DrawerContent>
    </Drawer>;
};

DetailWrapper.propTypes = {
    children: PropTypes.any,
    hideInvLink: PropTypes.bool,
    showTags: PropTypes.bool,
    appName: PropTypes.oneOf([
        'general_information',
        'advisor',
        'insights',
        'compliance',
        'vulnerabilities',
        'patch'
    ]),
    className: PropTypes.string,
    Wrapper: PropTypes.elementType,
    hasAccess: PropTypes.bool,
    inventoryId: PropTypes.string.isRequired
};

DetailWrapper.defaultProps = {
    appName: 'general_information'
};

export default DetailWrapper;
