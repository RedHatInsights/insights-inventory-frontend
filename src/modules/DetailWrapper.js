import React from 'react';
import AsyncInventory from './AsyncInventory';
import DetailWrapperCmp from '../components/InventoryDetail/DetailRenderer';

const DetailWrapper = (props) => <AsyncInventory {...props} component={DetailWrapperCmp} />;

export default DetailWrapper;
