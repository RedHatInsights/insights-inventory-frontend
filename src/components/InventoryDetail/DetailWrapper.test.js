import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { createPromise as promiseMiddleware } from 'redux-promise-middleware';
import { mock } from '../../__mocks__/systemIssues';
import { BasicInfo } from '../InventoryDetailDrawer';
import DetailWrapper from './DetailWrapper';

jest.mock('../InventoryDetailDrawer', () => ({
  ...jest.requireActual('../InventoryDetailDrawer'),
  BasicInfo: jest.fn(),
}));

describe('DetailWrapper', () => {
  mock.onGet('/api/patch/v3/systems/test-id').reply(200, 'test');
  mock.onGet('/api/insights/v1/system/test-id/reports/').reply(200, 'test');
  mock
    .onGet(
      '/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=2'
    )
    .reply(200, 'low-test');
  mock
    .onGet(
      '/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=4'
    )
    .reply(200, 'moderate-test');
  mock
    .onGet(
      '/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=5'
    )
    .reply(200, 'important-test');
  mock
    .onGet(
      '/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=7'
    )
    .reply(200, 'critical-test');
  mock
    .onGet(
      '/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=2'
    )
    .reply(500);
  mock.onPost('/api/compliance/graphql').reply(200, 'test');
  let initialState;
  let mockStore;
  const _Date = Date;
  const currDate = new Date('1970');
  beforeAll(() => {
    /*eslint no-global-assign:off*/
    Date = class extends Date {
      constructor(...props) {
        if (props.length > 0) {
          return new _Date(...props);
        }

        return currDate;
      }

      static now() {
        return new _Date('1970').getTime();
      }
    };
  });

  afterAll(() => {
    Date = _Date;
  });
  beforeEach(() => {
    mockStore = configureStore([promiseMiddleware()]);
    initialState = {
      entityDetails: {
        loaded: true,
        entity: {
          id: 'some-id',
          updated: new Date(),
          culled_timestamp: new Date(),
          stale_warning_timestamp: new Date(),
          stale_timestamp: new Date(),
        },
        isToggleOpened: true,
      },
    };
  });

  describe('DOM', () => {
    beforeAll(() => {
      BasicInfo.mockImplementation(
        jest.requireActual('../InventoryDetailDrawer').BasicInfo
      );
    });

    it('should render without data', () => {
      const store = mockStore({
        entityDetails: {},
      });
      const view = render(
        <Provider store={store}>
          <DetailWrapper />
        </Provider>
      );

      expect(view.asFragment()).toMatchSnapshot();
    });

    it('should render with data', () => {
      const store = mockStore(initialState);
      const view = render(
        <Provider store={store}>
          <DetailWrapper />
        </Provider>
      );

      expect(view.asFragment()).toMatchSnapshot();
    });

    it('should pass correct props to BasicInfo', () => {
      BasicInfo.mockImplementation(() => <span>BasicInfo</span>);
      const store = mockStore(initialState);
      render(
        <Provider store={store}>
          <DetailWrapper />
        </Provider>
      );

      expect(BasicInfo).toHaveBeenCalledWith(
        {
          hideInvLink: undefined,
          showTags: undefined,
        },
        {} // ref
      );
    });

    it('should render Wrapper', () => {
      const store = mockStore(initialState);
      render(
        <Provider store={store}>
          <DetailWrapper Wrapper={() => <h2 aria-label="test">something</h2>} />
        </Provider>
      );

      expect(screen.getByRole('heading', { name: 'test' })).toHaveTextContent(
        'something'
      );
    });

    it('should render disabled insights icon when no insights_id', () => {
      initialState = {
        entityDetails: {
          loaded: true,
          entity: {
            id: 'some-id',
            updated: new Date(),
            culled_timestamp: new Date(),
            stale_warning_timestamp: new Date(),
            stale_timestamp: new Date(),
            per_reporter_staleness: {},
          },
          isToggleOpened: true,
        },
      };
      const store = mockStore(initialState);
      render(
        <Provider store={store}>
          <DetailWrapper />
        </Provider>
      );

      expect(screen.getByLabelText('Disconnected indicator')).toBeVisible();
    });

    it('should not render disabled isnights icon when insights_id', () => {
      initialState = {
        entityDetails: {
          loaded: true,
          entity: {
            id: 'some-id',
            updated: new Date(),
            culled_timestamp: new Date(),
            stale_warning_timestamp: new Date(),
            stale_timestamp: new Date(),
            per_reporter_staleness: {
              puptoo: {
                stale_timestamp: new Date(),
              },
            },
          },
          isToggleOpened: true,
        },
      };
      const store = mockStore(initialState);
      render(
        <Provider store={store}>
          <DetailWrapper />
        </Provider>
      );

      expect(
        screen.queryByLabelText('Disconnected indicator')
      ).not.toBeInTheDocument();
    });

    it('should render children', () => {
      const store = mockStore(initialState);
      render(
        <Provider store={store}>
          <DetailWrapper>
            <h2 aria-label="test">something</h2>
          </DetailWrapper>
        </Provider>
      );

      expect(screen.getByRole('heading', { name: 'test' })).toHaveTextContent(
        'something'
      );
    });

    it('should calculate classnames', () => {
      const store = mockStore(initialState);
      render(
        <Provider store={store}>
          <DetailWrapper className="test-classname" />
        </Provider>
      );

      expect(screen.getByTestId('inventory-drawer')).toHaveClass(
        'test-classname'
      );
    });
  });

  describe('API', () => {
    it('should call open', async () => {
      const store = mockStore(initialState);
      render(
        <Provider store={store}>
          <DetailWrapper className="test" />
        </Provider>
      );

      await userEvent.click(
        screen.getByRole('button', {
          name: /close drawer panel/i,
        })
      );
      const actions = store.getActions();
      expect(actions[actions.length - 1]).toMatchObject({
        payload: { isOpened: false },
        type: 'TOGGLE_INVENTORY_DRAWER',
      });
    });
  });
});
