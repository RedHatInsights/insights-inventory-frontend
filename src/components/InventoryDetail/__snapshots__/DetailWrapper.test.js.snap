// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailWrapper DOM should render with data 1`] = `
<DocumentFragment>
  <div
    class="pf-c-drawer pf-m-expanded ins-c-inventory__drawer "
    data-testid="inventory-drawer"
  >
    <div
      class="pf-c-drawer__main"
    >
      <div
        class="pf-c-drawer__content"
      >
        <div
          class="pf-c-drawer__body"
        />
      </div>
      <div
        class="pf-c-drawer__panel"
        id="pf-drawer-panel-1"
      >
        <div
          class="pf-c-drawer__body"
        >
          <div
            class="pf-c-drawer__head"
          >
            <div
              class="pf-l-stack pf-m-gutter"
            >
              <div
                class="pf-l-stack__item"
              >
                <div
                  class="pf-l-split"
                >
                  <div
                    class="pf-l-split__item pf-m-fill"
                  >
                    <h3
                      class="pf-c-title pf-m-xl"
                      data-ouia-component-id="OUIA-Generated-Title-1"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe="true"
                    />
                  </div>
                  <div
                    class="pf-l-split__item"
                  >
                    <a
                      class="ins-c-entity-detail__inv-link"
                      href="./insights/inventory/some-id"
                    >
                      Open in Inventory
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="pf-c-drawer__actions"
            >
              <div
                class="pf-c-drawer__close"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close drawer panel"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-c-drawer__body"
        >
          <div
            class="pf-l-stack ins-c-inventory__drawer--content"
          >
            <div
              class="pf-l-stack__item"
            >
              <div
                class="pf-l-stack pf-m-gutter ins-c-inventory__drawer-system-issues"
              >
                <div
                  class="pf-l-stack__item"
                >
                  <h4
                    class="pf-c-title pf-m-lg"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Top system issues
                  </h4>
                </div>
                <div
                  class="pf-l-stack__item"
                >
                  <div
                    class="pf-l-level"
                  >
                    <div>
                      <div
                        class="pf-l-stack pf-m-gutter"
                      >
                        <div
                          class="pf-l-stack__item ins-c-inventory__drawer--title"
                        >
                          Applicable CVEs
                        </div>
                        <div
                          class="pf-l-stack__item"
                        >
                          <div
                            class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                          >
                            <span
                              class="pf-u-screen-reader"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div>
                      <div
                        class="pf-l-stack pf-m-gutter"
                      >
                        <div
                          class="pf-l-stack__item ins-c-inventory__drawer--title"
                        >
                          Applicable advisories
                        </div>
                        <div
                          class="pf-l-stack__item"
                        >
                          <div
                            class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                          >
                            <span
                              class="pf-u-screen-reader"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="pf-l-stack__item"
                >
                  <div
                    class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                  >
                    <span
                      class="pf-u-screen-reader"
                    />
                  </div>
                </div>
                <div
                  class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__md"
                >
                  <span
                    class="pf-u-screen-reader"
                  />
                </div>
              </div>
            </div>
            <div
              class="pf-l-stack__item pf-m-fill ins-c-inventory__drawer--facts"
            >
              <div
                class="pf-l-grid ins-entity-facts"
              >
                <div
                  class="pf-l-grid__item pf-m-6-col-on-md"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class=""
                    >
                      UUID:
                    </div>
                    <div
                      class="pf-m-grow"
                    >
                      some-id
                    </div>
                  </div>
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class=""
                    >
                      Last seen:
                    </div>
                    <div
                      class="pf-m-grow"
                    >
                      <span
                        class="ins-c-inventory__culling-danger"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                          />
                        </svg>
                        01 Jan 1970 00:00 UTC
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DetailWrapper DOM should render without data 1`] = `
<DocumentFragment>
  <div
    class="pf-c-drawer ins-c-inventory__drawer "
    data-testid="inventory-drawer"
  >
    <div
      class="pf-c-drawer__main"
    >
      <div
        class="pf-c-drawer__content"
      >
        <div
          class="pf-c-drawer__body"
        />
      </div>
      <div
        class="pf-c-drawer__panel"
        hidden=""
        id="pf-drawer-panel-0"
      />
    </div>
  </div>
</DocumentFragment>
`;
